packages/smart-contracts/src/contracts/BatchPayments.sol

```diff
@@ -12,56 +13,56 @@ import '@openzeppelin/contracts/security/ReentrancyGuard.sol';
+ *                  An additional batch fee is paid to the same address.
+ *         If one transaction of the batch fail, every transactions are reverted.
  */
 contract BatchPayments is Ownable, ReentrancyGuard {
-  // This event is declared in ERC20FeeProxy
-  event TransferWithReferenceAndFee(
-    address tokenAddress,
-    address to,
-    uint256 amount,
-    bytes indexed paymentReference,
-    uint256 feeAmount,
-    address feeAddress
-  );
+  using SafeERC20 for IERC20;
+
+  IERC20FeeProxy public paymentErc20FeeProxy;
+  IEthereumFeeProxy public paymentEthFeeProxy;
 
   // @dev: Between 0 and 1000, i.e: batchFee = 10 represent 1% of fee
   uint256 public batchFee;
 
-  IERC20FeeProxy public paymentErc20FeeProxy;
-  IEthereumFeeProxy public paymentEthereumFeeProxy;
+  struct Token {
+    address tokenAddress;
+    uint256 amountAndFee;
+    uint256 batchFeeAmount;
+  }
 
   /**
    * @param _paymentErc20FeeProxy The address to the ERC20 payment proxy to use.
-   * @param _paymentEthereumFeeProxy The address to the Ethereum payment proxy to use.
+   * @param _paymentEthFeeProxy The address to the Ethereum payment proxy to use.
    * @param _owner Owner of the contract.
    */
   constructor(
     address _paymentErc20FeeProxy,
-    address _paymentEthereumFeeProxy,
+    address _paymentEthFeeProxy,
     address _owner
   ) {
     paymentErc20FeeProxy = IERC20FeeProxy(_paymentErc20FeeProxy);
-    paymentEthereumFeeProxy = IEthereumFeeProxy(_paymentEthereumFeeProxy);
-    batchFee = 0;
+    paymentEthFeeProxy = IEthereumFeeProxy(_paymentEthFeeProxy);
     transferOwnership(_owner);
+    batchFee = 0;
   }
 
-  // Needed because batchEthPaymentsWithReferenceAndFee requires that the contract has ethers
-  receive() external payable {}
+  // batch Eth requires batch contract to receive funds from ethFeeProxy
+  receive() external payable {
+    require(msg.value == 0, 'Non-payable');
+  }
 
   /**
-   * @notice Send a batch of Ethereum payments w/fees with paymentReferences to multiple accounts.
-   *         The sum of _amounts and _feeAmounts must be <= to msg.value.
+   * @notice Send a batch of Eth payments w/fees with paymentReferences to multiple accounts.
    *         If one payment failed, the whole batch is reverted
    * @param _recipients List of recipients accounts.
    * @param _amounts List of amounts, corresponding to recipients[].
    * @param _paymentReferences List of paymentRefs, corr. to the recipients[].
    * @param _feeAmounts List of amounts of the payment fee, corr. to the recipients[].
    * @param _feeAddress The fee recipient.
-   * @dev Uses EthereumFeeProxy.sol to pay an invoice and fees, with a payment reference.
-   *      Make sure: msg.value >= sum(_amouts)+sum(_feeAmounts)
+   * @dev It uses EthereumFeeProxy to pay an invoice and fees, with a payment reference.
+   *      Make sure: msg.value >= sum(_amouts)+sum(_feeAmounts)+sumBatchFeeAmount
```

Comment:
There are many occurrences of `amout` accross files, can you fix them first please?